pkg/vm/engine/disttae/txn.go

```diff
@@ -140,9 +142,103 @@ func (txn *Transaction) WriteBatch(
 		return err
 	}
 
+	txn.DumpBatch(false)
+
 	return nil
 }
 
+func (txn *Transaction) DumpBatch(force bool) error {
+	// if txn.workspaceSize >= colexec.WriteS3Threshold {
+	if txn.workspaceSize >= colexec.WriteS3Threshold || force && txn.workspaceSize >= colexec.TagS3Size {
+		mp := make(map[[2]string][]*batch.Batch)
+		for i := 0; i < len(txn.writes); i++ {
+			idx := -1
+			for j := 0; j < len(txn.writes[i]); j++ {
+				if txn.writes[i][j].typ == INSERT && txn.writes[i][j].fileName == "" {
+					key := [2]string{txn.writes[i][j].databaseName, txn.writes[i][j].tableName}
+					bat := txn.writes[i][j].bat
+					// skip rowid
+					bat.Attrs = bat.Attrs[1:]
+					bat.Vecs = bat.Vecs[1:]
+					mp[key] = append(mp[key], bat)
+				} else {
+					txn.writes[i][idx+1] = txn.writes[i][j]
+					idx++
+				}
+			}
+			txn.writes[i] = txn.writes[i][:idx+1]
+		}
+		for key := range mp {
+			container, tbl, err := txn.getContainer(key)
+			if err != nil {
+				return err
+			}
+			container.InitBuffers(mp[key][0], 0)
+			for i := 0; i < len(mp[key]); i++ {
+				container.Put(mp[key][i], 0)
+			}
+			old_ctx := txn.proc.Ctx
+			txn.proc.Ctx = context.Background()
```

Comment:
Why do you need this?

Reply:
fix context cancel bug

Reply:
This must be a bug, the upper layer passed context should not be in this case timeout, far from the timeout time, and did not exit.

Reply:
#8328，this will be fixed by people, shoule be pengzheng